packages/react-app/src/pages/details.tsx

```diff
@@ -0,0 +1,42 @@
+import './details.scss';
+
+import { createContext } from 'react';
+import { useLoaderData } from 'react-router-dom';
+
+import { Stack } from '@mui/material';
+
+import GiveawayAsideContent from '../components/GiveawayAsideContent';
+import GiveawayMainContent from '../components/GiveawayMainContent';
+import SubHeader from '../components/SubHeader';
+import { Giveaway } from '../lib/types';
+import DuckdartClient from '../services/backend';
+
+const GiveawayContext = createContext<Giveaway | null>(null);
+
+const loader = async ({ params }: any) => {
+  const giveaway = await DuckdartClient.getGiveaway(params.giveawayId);
```

Comment:
🦆 also, I would suggest something like `ApiClient` since the name might change in the future
```suggestion
  const giveaway = await ApiClient.getGiveaway(params.giveawayId);
```